notebooks/data-sources/TestGrid/metrics/metric_template.ipynb

```diff
@@ -315,17 +315,21 @@
     "                            # we are only considering the 'Overall' test and fetching the\n",
     "                            # time duration values for this test and setting it to 'None'\n",
     "                            # for all the other tests in each grid\n",
-    "                            if tests[i] == \"Overall\":\n",
+    "                            if overall_only:\n",
+    "                                if tests[i] == \"Overall\":\n",
+    "                                    try:\n",
+    "                                        test_duration = graphs[i][0][\"values\"][0][j]\n",
+    "                                    except IndexError:\n",
+    "                                        test_duration = None\n",
```

Comment:
You need another `else:` statement here to handle updating the value of `test_duration` to `None` when the test name is not "Overall".  See lines 7 & 8 below.  
```
                           if overall_only:
                                if tests[i] == "Overall":
                                    try:
                                        test_duration = graphs[i][0]["values"][0][j]
                                    except IndexError:
                                        test_duration = None
                                else:
                                    test_duration = None
                            else:
                                try:
                                    graphs[i][0].keys()
                                    try:
                                        graphs[i][0]["values"][0][j]
                                        test_duration = graphs[i][0]["values"][0][j]
                                    except IndexError:
                                        test_duration = None
                                except TypeError:
                                    test_duration = None
```

The current approach just applies the same incorrect value to every row after the last "Overall" entry.

Reply:
Thanks for pointing that out. Fixed it :+1: